# scipy.special.geterr

> 原文链接：[https://docs.scipy.org/doc/scipy-1.12.0/reference/generated/scipy.special.geterr.html#scipy.special.geterr](https://docs.scipy.org/doc/scipy-1.12.0/reference/generated/scipy.special.geterr.html#scipy.special.geterr)

```py
scipy.special.geterr()
```

获取当前特殊函数错误处理的方式。

返回：

**err**字典

一个包含键“singular”、“underflow”、“overflow”、“slow”、“loss”、“no_result”、“domain”、“arg”和“other”的字典，它们的值是字符串“ignore”、“warn”和“raise”。这些键表示可能的特殊函数错误，而值定义了如何处理这些错误。

另请参阅

[`seterr`](scipy.special.seterr.html#scipy.special.seterr "scipy.special.seterr")

设置特殊函数错误的处理方式

[`errstate`](scipy.special.errstate.html#scipy.special.errstate "scipy.special.errstate")

特殊函数错误处理的上下文管理器

[`numpy.geterr`](https://numpy.org/devdocs/reference/generated/numpy.geterr.html#numpy.geterr "(在 NumPy v2.0.dev0)")

浮点错误的类似numpy函数

注意事项

完整文档有关特殊函数错误类型及处理选项，请参见[`seterr`](scipy.special.seterr.html#scipy.special.seterr "scipy.special.seterr")。

示例

默认情况下所有错误均被忽略。

```py
>>> import scipy.special as sc
>>> for key, value in sorted(sc.geterr().items()):
...     print("{}: {}".format(key, value))
...
arg: ignore
domain: ignore
loss: ignore
no_result: ignore
other: ignore
overflow: ignore
singular: ignore
slow: ignore
underflow: ignore 
```
